rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/server/NSCreationServlet.java

```diff
@@ -96,44 +102,58 @@ private void createNs(HttpServletRequest request, HttpServletResponse response)
                             deployNodeSource = true;
                         }
-                    } else if (fieldName.equals("infra")) {
-                        infra = fi.getString();
+                    } else if (formFieldName.equals("nodeSourceEdited")) {
+                        nodeSourceEdited = Boolean.valueOf(formFieldValue);
+                    } else if (formFieldName.equals("infra")) {
+                        infra = formFieldValue;
                         readingInfraParams = true;
-                    } else if (fieldName.equals("policy")) {
-                        policy = fi.getString();
+                    } else if (formFieldName.equals("policy")) {
+                        policy = formFieldValue;
                         readingPolicyParams = true;
                         readingInfraParams = false;
                     } else if (readingInfraParams) {
-                        infraParams.add(fi.getString());
+                        if (formFieldName.endsWith(NodeSourceEditWindow.EDIT_OR_UPLOAD_FORM_ITEM_SUFFIX)) {
+                            takeInlineContent = formFieldValue.endsWith(NodeSourceEditWindow.EDIT_RADIO_OPTION_NAME);
+                        } else if (formFieldName.endsWith(NodeSourceEditWindow.EDIT_FORM_ITEM_SUFFIX)) {
+                            if (takeInlineContent) {
+                                infraFileParams.add(formFieldValue);
+                            }
+                        } else {
+                            infraParams.add(formFieldValue);
+                        }
                     } else if (readingPolicyParams) {
-                        policyParams.add(fi.getString());
+                        policyParams.add(formFieldValue);
                     } else {
-                        LOGGER.warn("Unexpected param " + fieldName);
+                        LOGGER.warn("Unexpected param " + formFieldName);
                     }
                 } else {
                     if (readingInfraParams) {
-                        byte[] bytes = IOUtils.toByteArray(fi.getInputStream());
-                        infraFileParams.add(new String(bytes));
+                        if (!takeInlineContent) {
+                            byte[] bytes = IOUtils.toByteArray(formField.getInputStream());
+                            infraFileParams.add(new String(bytes));
+                            takeInlineContent = true;
+                        }
                     } else if (readingPolicyParams) {
-                        byte[] bytes = IOUtils.toByteArray(fi.getInputStream());
+                        byte[] bytes = IOUtils.toByteArray(formField.getInputStream());
                         policyFileParams.add(new String(bytes));
                     } else {
-                        LOGGER.warn("Unexpected param " + fieldName);
+                        LOGGER.warn("Unexpected param " + formFieldName);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use the built-in formatting to construct this argument. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2629)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Format specifiers should be used instead of string concatenation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3457)